### DISCUSSION TITLE
How to convert byte[] to a .jpeg image with ImageSharp?

### DISCUSSION THREAD

**Discussion by Acesteef** [SixLabors/ImageSharp#1408] [Q&A] [answered]
I have my image stored in my database as VARBINARY(MAX) then i convert it into a byte[]. Now i want to be able to display the image in my front-end as a .jpeg image. I tried converting the byte[] to a .jpeg with the `.SaveAsJpeg()` method but its still a `Image<Rgba32>`. I know i am doing something wrong, i just dont know what. I hope someone can give me a code example of how it is actually done.

This is my code: 
 
```
 public Image ConvertByteArray()
 {
                           
            Stream outputStream = new MemoryStream();

            var image = Image.Load<Rgba32>(this.MoviePoster); 

            image.SaveAsJpeg(outputStream, new JpegEncoder() { Quality = 100 });

            return image;
 }
```

**Comment by JimBobSquarePants** [comment, marked as answer]
> I tried converting the byte[] to a .jpeg with the .SaveAsJpeg() method but its still a Image<Rgba32>. I know i am doing something wrong, i just dont know what. I hope someone can give me a code example of how it is actually done.

I cannot actually understand what you want to do. You have an encoded image in the database so why do you need to load it with ImageSharp?

**Reply by Acesteef** [reply]
I need to show the image in the front-end, so i need to find a way to convert the `VARBINARY` to a format that i can display in my html. Im not a very experienced programmer yet so i dont really know the best way to do this. I thought ImageSharp would allow me to do this?

**Reply by antonfirsov** [reply]
@Acesteef I see two problems here:
1. Understanding what does the `VARBINARY` blob actually store. If it's already a  jpeg-encoded blob, you don't need ImageSharp, it might be just enough to read the `VARBINARY` content and copy it to the output stream.
2. Assuming you actually need ImageSharp (maybe because you need to convert to jpeg from another format, or you want to resize the image before sending it to the frontend etc.):
After the call to `image.SaveAsJpeg(outputStream, new JpegEncoder() { Quality = 100 });` your image is encoded in `outputStream` in Jpeg format. You need to somehow copy the contents of this stream to your client. From an ASP.NET Core Controller you can use `File(outputStream,  "image/jpeg")`
